### PR TITLE
BUG: added 2 missing lines

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by seanm** [comment]
@thewtex here are the 2 missing lines you requested in #3037.  I have no idea what this does or fixes though, so could you propose a proper commit message and I'll amend the commit...

**Comment by dzenanz** [comment]
I remember there was some discussion about this not long ago. @N-Dekker @hjmjohnson

**Comment by dzenanz** [comment]
This is the PR where that somewhat related discussion happened: #4485.

**Comment by thewtex** [comment]
@seanm per #3037 , those lines are meant to go at the start of 

```
template <typename TScalar, unsigned int NInputDimensions,
          unsigned int NOutputDimensions>
MatrixOffsetTransformBase<TScalar, NInputDimensions, NOutputDimensions>
::MatrixOffsetTransformBase(const MatrixType & matrix, const OutputVectorType & offset)
{
// here
```

**Comment by thewtex** [comment]
And it may not be correct following the discussion around:

https://github.com/InsightSoftwareConsortium/ITK/pull/4485#issuecomment-1975185951

**Comment by seanm** [comment]
> And it may not be correct

So shall I just close/discard this?

**Comment by N-Dekker** [comment]
> Ideally, Hans and/or Niels will review this

OK, thanks @dzenanz I'll have a look!  (I'm back 😃!)

**Comment by N-Dekker** [comment]
I'm still looking for an actual relevant use case for the `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)` constructor. It's protected, and it isn't being used by `MatrixOffsetTransformBase::New()`, obviously. Is it being tested, directly or indirectly?

File: Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx [hunk 1]
```diff
@@ -48,6 +48,8 @@ MatrixOffsetTransformBase<TParametersValueType, VInputDimension, VOutputDimensio
   m_MatrixMTime.Modified();
   std::copy_n(offset.begin(), VOutputDimension, m_Translation.begin());
   this->ComputeMatrixParameters();
```
Comment (N-Dekker) [inline Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx:50]: Slightly off-topic, maybe, but the `this->ComputeMatrixParameters()` call does nothing at all: https://github.com/InsightSoftwareConsortium/ITK/blob/07f8459bdf913647dab52108d8ff284d05eb4fe4/Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx#L678-L684

(I was expecting `ComputeMatrixParameters()` to interact with `m_FixedParameters`, but obviously, it does not!)

**Comment by N-Dekker** [comment]
I think the unit test with this PR might be as follows:

```cpp
TEST(MatrixOffsetTransformBase, CreateWithMatrixAndOffset)
{
  class DerivedTransform : public itk::MatrixOffsetTransformBase<>
  {
  public:
    ITK_DISALLOW_COPY_AND_MOVE(DerivedTransform);

    static auto
    Create()
    {
      // Indirectly call the `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)` constructor.
      const itk::SmartPointer<DerivedTransform> ptr = new DerivedTransform(MatrixType(), OutputVectorType());
      ptr->UnRegister();
      return ptr;
    }

  private:
    // Inherit the constructors of MatrixOffsetTransformBase:
    using itk::MatrixOffsetTransformBase<>::MatrixOffsetTransformBase;
  };

  const auto                                  transform = DerivedTransform::Create();
  const DerivedTransform::FixedParametersType expectedFixedParameters(DerivedTransform::InputSpaceDimension, 0);

  EXPECT_EQ(transform->GetFixedParameters(), expectedFixedParameters);
}

```

The last line is the essential one here: `EXPECT_EQ(transform->GetFixedParameters(), expectedFixedParameters)` checks that `m_FixedParameters` has the expected number of zero-values. But moreover, it checks that `GetFixedParameters()` does not crash! Would that be helpful to you? (The test could be added to Core/Transform/test/itkMatrixOffsetTransformBaseGTest.cxx)

File: Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx (hunk 1 above)
Comment (N-Dekker) [reply Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx:50]: If this `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)` constructor is really meant to be used, I guess it should do the same as the default-constructor, so it should then also do `m_InverseMatrixMTime = m_MatrixMTime`, just like the default-constructor:

https://github.com/InsightSoftwareConsortium/ITK/blob/fb4b463b38640c874202808311ff6e06afb1237e/Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx#L30-L38

(But it does not need to call `ComputeMatrixParameters()` anyway.)

**Comment by hjmjohnson** [comment]
@N-Dekker Added test that you suggested.

**Review by thewtex — APPROVED** [review]
🙌 💯 

@seanm @N-Dekker @hjmjohnson awesome teamwork!

**Comment by N-Dekker** [comment]
Honestly, if this specific `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)` constructor isn't really useful, I would rather see it being removed. Would be willing to sacrifice the beautiful 😸 unit test that I proposed, with this PR.

**Comment by hjmjohnson** [comment]
\azp ITK.Linux

File: Modules/Core/Transform/include/itkMatrixOffsetTransformBase.h [hunk 2]
```diff
@@ -507,7 +507,10 @@ class ITK_TEMPLATE_EXPORT MatrixOffsetTransformBase
    * to values specified by the caller.  If the arguments are
    * omitted, then the MatrixOffsetTransformBase is initialized to an identity
    * transformation in the appropriate number of dimensions. */
-  MatrixOffsetTransformBase(const MatrixType & matrix, const OutputVectorType & offset);
+#if !defined(ITK_LEGACY_REMOVE)
+  [[deprecated("Removed unsued constructor")]] MatrixOffsetTransformBase(const MatrixType &       matrix,
```
Comment (N-Dekker) [inline Modules/Core/Transform/include/itkMatrixOffsetTransformBase.h:511]: Can you please fix this little typo here, "unsued"? Although I know it's not _that_ bad, we won't get _sued_  😸

**Comment by N-Dekker** [comment]
@hjmjohnson There are a few similar unused constructors in derived classes (11!) that should also be deprecated, when `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)` is deprecated:

    AffineTransform(const MatrixType & matrix, const OutputVectorType & offset);
    ComposeScaleSkewVersor3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    FixedCenterOfRotationAffineTransform(const MatrixType & matrix, const OutputVectorType & offset);
    QuaternionRigidTransform(const MatrixType & matrix, const OutputVectorType & offset);
    Rigid3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    ScalableAffineTransform(const MatrixType & matrix, const OutputVectorType & offset);
    ScaleSkewVersor3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    ScaleVersor3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    Similarity3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    VersorRigid3DTransform(const MatrixType & matrix, const OutputVectorType & offset);
    VersorTransform(const MatrixType & matrix, const OutputVectorType & offset);

These 11 constructors all directly or indirectly call `MatrixOffsetTransformBase(const MatrixType &, const OutputVectorType &)`. But none of them are tested, it appears that they aren't even compiled!!! At least not at the CI. And they are all protected, so they just won't be used anywhere, I guess.

**Comment by hjmjohnson** [comment]
@N-Dekker Thanks.  I have implemented your additional suggestions.

File: Modules/Core/Transform/include/itkMatrixOffsetTransformBase.h (hunk 2 above)
Comment (N-Dekker) [reply Modules/Core/Transform/include/itkMatrixOffsetTransformBase.h]: Thanks for fixing the typo 👍

**Pull request merged**